components/style/values/computed/image.rs

```diff
@@ -0,0 +1,462 @@
+pub enum GradientKind {
+    Linear(AngleOrCorner),
+    Radial(EndingShape, Position),
+}
+
+impl ToComputedValue for specified::GradientKind {
+    type ComputedValue = GradientKind;
+
+    #[inline]
+    fn to_computed_value(&self, context: &Context) -> GradientKind {
+        match *self {
+            specified::GradientKind::Linear(angle_or_corner) => {
+                GradientKind::Linear(angle_or_corner)
+            },
+            specified::GradientKind::Radial(ref shape, position) => {
+                GradientKind::Radial(shape.to_computed_value(context),
+                                     position.to_computed_value(context))
+            },
+        }
+    }
+    #[inline]
+    fn from_computed_value(computed: &GradientKind) -> Self {
+        match *computed {
+            GradientKind::Linear(angle_or_corner) => {
+                specified::GradientKind::Linear(ToComputedValue::from_computed_value(&angle_or_corner))
+            },
+            GradientKind::Radial(ref shape, position) => {
+                specified::GradientKind::Radial(ToComputedValue::from_computed_value(shape),
+                                                ToComputedValue::from_computed_value(&position))
+            },
+        }
+    }
+}
+
+/// Computed values for one color stop in a linear gradient.
+/// https://drafts.csswg.org/css-images/#typedef-color-stop-list
+#[derive(Clone, PartialEq, Copy)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub struct ColorStop {
+    /// The color of this stop.
+    pub color: CSSColor,
+
+    /// The position of this stop. If not specified, this stop is placed halfway between the
+    /// point that precedes it and the point that follows it per CSS-IMAGES § 3.4.
+    pub position: Option<LengthOrPercentage>,
+}
+
+impl ::cssparser::ToCss for ColorStop {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        try!(self.color.to_css(dest));
+        if let Some(position) = self.position {
+            try!(dest.write_str(" "));
+            try!(position.to_css(dest));
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Debug for ColorStop {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let _ = write!(f, "{:?}", self.color);
+        self.position.map(|pos| {
+            let _ = write!(f, " {:?}", pos);
+        });
+        Ok(())
+    }
+}
+
+impl ToComputedValue for Vec<specified::ColorStop> {
+    type ComputedValue = Vec<ColorStop>;
+
+    #[inline]
+    fn to_computed_value(&self, context: &Context) -> Vec<ColorStop> {
+        self.iter().map(|stop| {
+            ColorStop {
+                color: stop.color.parsed,
+                position: match stop.position {
+                    None => None,
+                    Some(value) => Some(value.to_computed_value(context)),
+                },
+            }
+        }).collect()
+    }
+    #[inline]
+    fn from_computed_value(computed: &Vec<ColorStop>) -> Self {
```

Comment:
Er, handle the vec. Basically, just do `stops.map(ToComputedValue::to_computed_value)` or something